containers-tests/tests/intset-properties.hs

```diff
@@ -141,6 +143,16 @@ prop_InsertIntoEmptyValid :: Int -> Property
 prop_InsertIntoEmptyValid x =
     valid (insert x empty)
 
+{--------------------------------------------------------------------
+  Instances for Eq and Ord
+--------------------------------------------------------------------}
+
+prop_instanceEqIntSet :: IntSet -> IntSet -> Bool
+prop_instanceEqIntSet x y = (x == y) == (toAscList x == toAscList y)
+
+prop_instanceOrdIntSet :: IntSet -> IntSet -> Bool
+prop_instanceOrdIntSet x y = (compare x y) == (compare (toAscList x) (toAscList y))
```

Comment:
We already have `prop_ord` for this. I don't see a similar `prop_eq` though.

Reply:
not exactly "this" (`prop_ord` uses `toList`, I think it should use `toAscList` https://github.com/haskell/containers/issues/470#issuecomment-513856562 )

Reply:
Ah, yes, `toAscList` is a bit more correct. That said, I would be astonished if we made `toList` do something different. While it's true that it's not documented as producing keys in order, it's also not explicitly documented as potentially producing keys in some other order. There are almost certainly people whose code will break in horrible ways if we change that.